x-pack/plugins/security_solution/public/detections/components/rules/step_define_rule/index.tsx

```diff
@@ -165,7 +166,20 @@ const StepDefineRuleComponent: FC<StepDefineRuleProps> = ({
   // reset form when rule type changes
   useEffect(() => {
     reset({ resetValues: false });
-  }, [reset, ruleType]);
+    if (getFields().queryBar != null) {
+      const queryBar = getFields().queryBar.value as DefineStepRule['queryBar'];
+      if (queryBar.edited !== true && !isUpdateView) {
+        if (isThreatMatchRule(ruleType)) {
+          setFieldValue('queryBar', {
```

Comment:
`field.setValue()` has a format similar to `setState` where it accepts a function that receives the previous value; that might be more appropriate than merging in the original defaults, here.

Reply:
Thanks, see comments above, I can use this in combination with reset and I think we are going to get what we want out of this feature in a more straight forward way.